lib/awful/util.lua

```diff
@@ -110,6 +112,61 @@ function util.spawn_with_shell(cmd, sn)
     end
 end
 
+--- Spawn a program and asynchronously capture its output line by line.
+-- @tparam string|table cmd The command.
+-- @tparam function line_callback Function to call with each line of output.
```

Comment:
Please document the signature of the callback function, e.g. that it gets `line_from_stdout`.